src/Data/NaturalTransformation.purs

```diff
@@ -0,0 +1,14 @@
+module Data.NaturalTransformation where
+
+-- | A type for natural transformations between functors (although the `Functor`
+-- | constraint is not enforced).
+-- |
+-- | A natural transformation allows a functor to be transformed into another
+-- | without changing the internal structure. This is a little similar to the
+-- | `map` operation of a `Functor`, although "one level up". In the case of
+-- | `map`, the functor is preserved while the inner values are re-mapped,
+-- | whereas here the inner values are preserved and the functor is changed
+-- | instead.
+type NaturalTransformation f g = forall a. f a -> g a
+
+infixr 4 type NaturalTransformation as ~>
```

Comment:
I figured this gives us enough room to slide `->` underneath while still not making it the lowest precedence, when we decide what that should be.